crates/runtime/src/flight/do_put.rs

```diff
@@ -16,32 +16,40 @@ limitations under the License.
 
 use std::{collections::HashMap, sync::Arc};
 
-use arrow_flight::{flight_service_server::FlightService, FlightData, PutResult};
+use arrow::array::RecordBatch;
+use arrow_flight::{
+    flight_service_server::FlightService, utils::flight_data_to_arrow_batch, FlightData, PutResult,
+};
 use arrow_ipc::convert::try_schema_from_flatbuffer_bytes;
-use datafusion::sql::TableReference;
-use futures::stream;
-use tokio::sync::{broadcast::Sender, RwLock};
+use datafusion::{
+    error::DataFusionError, execution::SendableRecordBatchStream,
+    physical_plan::stream::RecordBatchStreamAdapter, sql::TableReference,
+};
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::ReceiverStream;
 use tonic::{Request, Response, Status, Streaming};
 
+use async_stream::stream;
+
 use crate::{
-    dataupdate::{DataUpdate, UpdateType},
+    dataupdate::{StreamingDataUpdate, UpdateType},
     timing::TimedStream,
 };
 
 use super::{metrics, Service};
 
-async fn get_sender_channel(
-    channel_map: Arc<RwLock<HashMap<TableReference, Arc<Sender<DataUpdate>>>>>,
-    path: &TableReference,
-) -> Option<Arc<Sender<DataUpdate>>> {
-    let channel_map_read = channel_map.read().await;
-    if channel_map_read.contains_key(path) {
-        let channel = channel_map_read.get(path)?;
-        Some(Arc::clone(channel))
-    } else {
-        None
-    }
-}
+// async fn get_sender_channel(
```

Comment:
Investigating if this is still required and non-streaming update should be kept as-is specifically for this case.